src/three-components/SmoothControls.ts

```diff
@@ -0,0 +1,533 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Camera, EventDispatcher, Quaternion, Spherical, Vector2, Vector3} from 'three';
+
+import {clamp, step} from '../utils.js';
+
+export type EventHandlingBehavior = 'prevent-all'|'prevent-handled';
+export type ZoomPolicy = 'always-allow'|'allow-when-focused';
+export type TouchMode = 'rotate'|'zoom';
+
+export interface SmoothControlsConstraints {
```

Comment:
I'm not sure of the word "constraints" (and below, e.g. `applyConstraints`), as it's accurate for most of the options, but less obvious for `acceleration`, etc.

Reply:
👍 I don't love it either, open to alternatives

Reply:
Maybe `configuration` or `options` or something, but don't feel strongly (and more important if used outside of model-viewer)

Reply:
Done